File: frontend/lib/models/zone.dart
class Zone {
  final int zoneID;
  final String name;
  final int capacity;
  final double pricePerHour;
  final int maxDuration;
  final String accessLevel;
  final String zoneType;

  const Zone({
    required this.zoneID,
    required this.name,
    required this.capacity,
    required this.pricePerHour,
    required this.maxDuration,
    required this.accessLevel,
    required this.zoneType,
  });

  factory Zone.fromJson(Map<String, dynamic> json) {
    return Zone(
      zoneID: json['zoneID'] as int,
      name: json['name'] as String,
      capacity: json['capacity'] as int,
      pricePerHour: (json['pricePerHour'] as num).toDouble(),
      maxDuration: json['maxDuration'] as int,
      accessLevel: json['accessLevel'] as String,
      zoneType: json['zoneType'] as String,
    );
  }
}

File: backend/Domain/Entities/Zone.cs
using SmartParking.Domain.Enums;

namespace SmartParking.Domain.Entities;

public class Zone
{
    public int ZoneID { get; set; }
    public string Name { get; set; } = string.Empty;
    public int Capacity { get; set; }
    public double PricePerHour { get; set; }
    public int MaxDuration { get; set; }
    public AccessLevel AccessLevel { get; set; }
    public ZoneType ZoneType { get; set; }
    public ZoneStatus Status { get; set; } = ZoneStatus.Active;
    public string GeoJson { get; set; } = string.Empty;

    public ICollection<ParkingSpot> Spots { get; set; } = new List<ParkingSpot>();
}
